common/src/lazy_pages.rs

```diff
@@ -20,20 +20,52 @@
 
 use crate::Origin;
 use core::convert::TryFrom;
+use core::fmt;
 use gear_core::ids::ProgramId;
 use gear_core::memory::{PageBuf, PageNumber};
-use gear_runtime_interface::gear_ri;
+use gear_runtime_interface::{gear_ri, GetReleasedPageError, MprotectError};
 use sp_std::{boxed::Box, collections::btree_map::BTreeMap, vec::Vec};
 
-fn mprotect_lazy_pages(addr: u64, protect: bool) -> Result<(), &'static str> {
-    gear_ri::mprotect_lazy_pages(addr, protect).map_err(|_| "Cannot mprotect some pages")
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Error {
+    Mprotect(MprotectError),
+    GetReleasedPage(GetReleasedPageError),
+    VecToPageData,
+    NoPageDataInStorage,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::Mprotect(err) => fmt::Display::fmt(err, f),
+            Error::GetReleasedPage(err) => fmt::Display::fmt(err, f),
+            Error::VecToPageData => write!(f, "Cannot convert vec to page data"),
+            Error::NoPageDataInStorage => write!(f, "Cannot find page data in storage"),
+        }
+    }
+}
+
+impl From<MprotectError> for Error {
+    fn from(err: MprotectError) -> Self {
+        Self::Mprotect(err)
+    }
+}
+
+impl From<GetReleasedPageError> for Error {
+    fn from(err: GetReleasedPageError) -> Self {
+        Self::GetReleasedPage(err)
+    }
+}
```

Comment:
what about using `thiserror` crate. It really helps with removing this.

Reply:
`common` crate uses `no_std` when `thiserror` relies on `std::error::Error`

Reply:
we can still remove that `Display` implementation with `#[derive(fmt = "")]`, isn't it?